deploy/terraform-custom-datacommons/README.md

```diff
@@ -0,0 +1,197 @@
+# Custom Data Commons Terraform Deployment
+
+## Overview
+
+Deploying your own Custom Data Commons instance on Google Cloud Platform (GCP) lets you to host and manage your own data while leveraging [Google's Data Commons](https://datacommons.org/) repository of publicly available information. This deployment is managed using Terraform, which automates the provisioning of infrastructure and services on GCP.
+
+## Features
+
+* Creates Data Commons Website service container in Cloud Run
+* Creates Data Commons Data task container in Cloud Run
+* Enables all required Google Cloud APIs
+* Creates Redis instance (optional)
+* Creates MySQL instance
+* Creates new service account with minimum required permissions
+* Automatically provisions required Data Commons API key. Stores key in GCP secrets container.
+* Automatically provisions required Data Commons API key. Stores key in GCP secrets container.
+* Generates random MySQL password and stores in GCP secrets container.
+* Supports multiple Data Commons instances in the same GCP account
+
+## Architecture
+
+![Custom Data Commons Architecture](https://docs.datacommons.org/assets/images/custom_dc/customdc_setup3.png)
+
+
+## Pre-requisites
+
+- **A [Google Cloud Project](https://cloud.google.com)**: You must have a GCP project where the custom Data Commons instance will be deployed.
+- **[gcloud](https://cloud.google.com/sdk/docs/install)**: Ensure gcloud is installed on your local machine for the initial environment setup.
+- **[Terraform](https://developer.hashicorp.com/terraform/install)**: Ensure Terraform is installed on your local machine to manage the infrastructure as code.
+
+## Deployment Instructions
+
+### 1. Configure Deployment Variables
+
+Create a local copy of `terraform.tfvars` file and fill in the required values.
+
+```
+cd deploy/terraform-custom-datacommons/
+cp ./modules/terraform.tfvars.sample ./modules/terraform.tfvars
+```
+
+Example `terraform.tfvars`:
+
+```
+project_id  = "your-gcp-project"
+namespace   = "dan-dev"
+dc_api_key  = "your-api-key"
+```
+
+#### Required Variables
+
+- **project_id**: The Google Cloud project ID where the resources will be created.
+- **namespace**: A unique namespace to differentiate multiple instances of custom Data Commons within the same project.
+- **dc_api_key**: Data Commons API key. [Request an API key](https://docs.google.com/forms/d/e/1FAIpQLSeVCR95YOZ56ABsPwdH1tPAjjIeVDtisLF-8oDYlOxYmNZ7LQ/viewform?resourcekey=0-yJ9nT9ST-TfoKNtmGIws-g)
+
+#### Optional Configuration Variables
+
+
+
+- **region**: The [GCP region](https://cloud.google.com/about/locations) where resources will be deployed.
+- **redis_enabled**: Set to false to disable redis caching.
```

Comment:
Just confirming we set this to true by default? If so, is that because that's the configuration we expect most partners will want?

Reply:
It's true by default, which is the configuration customers would want in production but probably not in development. Maybe it would make more sense to have it be false by default -wdyt?

Reply:
Changed the default to "false", with the rationale that users at first will be deploying to a development environment (and we don't have a good way to clear the redis cache right now)